packages/Syntax/src/globalStyles.tsx

```diff
@@ -0,0 +1,121 @@
+  '06': string; // Classes, Markup Bold, Search Text Background
+  '07': string; // Strings, Inherited Class, Markup Code, Diff Inserted
+  '08': string; // Support, Regular Expressions, Escape Characters, Markup Quotes
+  '09': string; // Functions, Methods, Classes, Names, Sections, Literals
+  '10': string; // Keywords, Storage, Selector, Markup Italic, Diff Changed
+}
+
+export const variantColors: { readonly [K in Variant]: Base16Palette } = {
+  [Variant.Light]: {
+    '00': uiColors.gray.light3,
+    '01': uiColors.gray.light2,
+    '02': uiColors.gray.dark2,
+    '04': uiColors.white,
+    '03': uiColors.gray.dark3,
+    '05': '#CA4821',
+    '06': '#EDB210',
+    '07': '#12824D',
+    '08': uiColors.blue.dark2,
+    '09': uiColors.blue.base,
+    '10': '#CC3887',
+  },
+
+  [Variant.Dark]: {
+    '00': uiColors.gray.dark3,
+    '01': uiColors.black,
+    '02': uiColors.gray.light1,
+    '03': uiColors.gray.light3,
+    '04': uiColors.gray.dark2,
+    '05': '#FF6F44',
+    '06': '#EDB210',
+    '07': '#35DE7B',
+    '08': '#a5e3ff',
+    '09': '#2DC4FF',
+    '10': '#FF7DC3',
+  },
+};
+
+const getStyles = (variant: Variant): string => `
+  .lg-highlight-hljs-${variant} {
+    .lg-highlight-quote,
+    .lg-highlight-literal,
+    .lg-highlight-class,
+    .lg-highlight-section,
+    .lg-highlight-name,
+    .lg-highlight-class > .lg-highlight-keyword,
+    .lg-highlight-function > .lg-highlight-keyword {
+      color: ${variantColors[variant]['09']};
+    }
+  
+    .lg-highlight-regexp,
+    .lg-highlight-params,
+    .lg-highlight-meta,
+    .lg-highlight-meta-string {
+      color: ${variantColors[variant]['08']};
+    }
+  
+    .lg-highlight-comment {
+      color: ${variantColors[variant]['02']};
+    }
+
+    .lg-highlight-doctag,
+    .lg-highlight-formula {
+      color: ${variantColors[variant]['03']};
+    }
+  
+    .lg-highlight-keyword,
+    .lg-highlight-selector-tag,
+    .lg-highlight-selector-attr,
+    .lg-highlight-selector-pseudo,
+    .lg-highlight-selector-id,
+    .lg-highlight-selector-class {
+      color: ${variantColors[variant]['10']};
+    }
+  
+    .lg-highlight-addition {
+      color: ${variantColors[variant]['04']};
+    }
+  
+    .lg-highlight-variable,
+    .lg-highlight-deletion,
+    .lg-highlight-symbol,
+    .lg-highlight-bullet,
+    .lg-highlight-subst,
+    .lg-highlight-meta,
+    .lg-highlight-link,
+    .lg-highlight-number,
+    .lg-highlight-attr,
+    .lg-highlight-attribute,
+    .lg-highlight-built_in,
+    .lg-highlight-template-variable,
+    .lg-highlight-type {
+      color: ${variantColors[variant]['05']}
+    }
+  
+    .lg-highlight-string,
+    .lg-highlight-title {
+      color: ${variantColors[variant]['07']};
+    }
+  
+    .lg-highlight-emphasis {
+      font-style: italic;
+    }
+  
+    .lg-highlight-strong {
+      font-weight: bold;
+    }
+  }
+`;
+
+Object.values(Variant).forEach(variant => injectGlobal(getStyles(variant)));
```

Comment:
That's a feature, not a 🐛haha

Had a feeling this might be controversial, but here's my logic and let me know if it's faulty: injectGlobal isn't a very clever implementation, and I believe if you have it in your render function, then it could inject duplicate stylesheets. We won't know what's being used until we're in the render function, so I found this to be the implementation that would be least disruptive.

The cherry on top: it's super nice to just do `import './globalStyles';` in the Syntax component file.

Reply:
So two concerns I have with this working this way:

- What if someone tests their application that uses this component in an environment without a simulated browser?
- We shouldn't add additional cost when loading an application if this component isn't visible.

If we had this behind a function and in the Syntax component had a global flag that said `let hasInsertedGlobalRules` and flip that to true after the first render then we'd defer inserting it until first used on the page.

Reply:
You'll need to describe what environment you're talking about. If it's SSR, Emotion handles this pretty well out the box (including global styles). If there's something else you're worried about, I don't think this is any more likely to break than our other usages of emotion.

I think I have something that solves for your issue here and in Syntax (registering languages).

Reply:
Commented where you made the modification. Having an import of a file shouldn't have side effects by default, as well as being harder to decouple that side effect at a later point.